### PR TITLE
Add docs to program initialization related code

### LINKED ISSUE #347: Submit code transaction
Since we want to initialise programs from other programs (see #245), one of the scenarios is to initialise program from existing code (by providing `code_hash` and initialization payload)

This `submit_code` transaction should put a code to the state and return its `code_hash`.

### PR DESCRIPTION
**Pull request by SabaunT**
In order to close #347, some research on how program initialization works was done. For the purpose of research `gtest` and underlying crates were discovered and (partly) documented in the context of program initialization. I didn't intend to document everything, but do that just in some bounded scope while understanding program submit process.

No issues were resolved here, actually.
Please merge asap.

### CODE REVIEW

**Review by shamilsan — APPROVED** [review]

File: core-runner/src/runner.rs [hunk 1]
```diff
@@ -533,10 +533,25 @@ impl<SC: StorageCarrier, E: Environment<Ext>> Runner<SC, E> {
         RunningContext::new(&self.config, allocations)
     }
 
-    /// Initialize new program.
+    /// Initialize a new program. This includes putting this program in the storage and dispatching
```
Comment (NikVolf) [inline core-runner/src/runner.rs:536]:
```suggestion
    /// Initialize a new program. 
    ///
    /// This includes putting this program in the storage and dispatching.
```

File: core-runner/src/runner.rs [hunk 2]
```diff
@@ -780,6 +813,8 @@ fn run<E: Environment<Ext>>(
     if let Some(max_page) = max_page {
         let max_page_num = *max_page.0;
         let mem_size = memory.size();
+        // If we charge for `max_page_num == memsize`, then we perform a double charge,
+        // because gas was charged previously. todo - check that by running tests
```
Comment (NikVolf) [inline core-runner/src/runner.rs:817]:
```suggestion
```

we don't need this kind of research in code comments, create an issue for that

File: gtest/src/js/mod.rs [hunk 3]
```diff
@@ -16,9 +16,22 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
+//! Module contains all the needed types and functionality for an easy inference of structures of messages sent to or
+//! received from *init* and *handle* functions..
```
Comment (NikVolf) [inline gtest/src/js/mod.rs:20]:
```suggestion
//! JS module.
//! 
//! Module contains all the needed types and functionality for an easy inference of structures of messages sent to or
//! received from *init* and *handle* functions..
```

**Comment by NikVolf** [comment]
I don't see any successful run of the CI on this

**Comment by SabaunT** [comment]
> I don't see any successful run of the CI on this
successful CI run on 945c2da commit

**Pull request merged**